pandas/core/arrays/arrow/array.py

```diff
@@ -330,6 +331,18 @@ def __getitem__(self, item: PositionalIndexer):
             else:
                 return scalar
 
+    def __iter__(self) -> Iterator[Any]:
+        """
+        Iterate over elements of the array.
+        """
+        na_value = self._dtype.na_value
+        for value in self._data:
```

Comment:
Is there any benefit to combining the chunks of `self._data` and then iterating?

Reply:
Maybe seeing a 5-10% improvement testing locally, but not much. Seems to be 5-10% with 10 chunks or 10000 chunks. I can add it if you want?

Reply:
I don't think I'm seeing a consistent improvement at low number of chunks. 

If curious, here is the snippet I'm testing with:

```
import pyarrow as pa

def time_iter(arr):
    for v in arr:
        pass

num_chunks = 2
chunk_size = 10**5
data = [np.arange(chunk_size)] * num_chunks
arr = pa.chunked_array(data)

%timeit time_iter(arr)
%timeit time_iter(arr.combine_chunks())
```

Reply:
IIUC doing combine_chunks will cause a copy, so any speed improvements will be offset by memory increase.  if so, i think not worth it